smart-contract-verifier/src/solidity/solc_cli.rs

```diff
@@ -0,0 +1,493 @@
+                .path()
+                .exists()
+                .then(|| &self.file_names)
+                .ok_or_else(|| {
+                    SolcError::Message("temp dir with contracts doesn't exists".to_string())
+                })
+        }
+    }
+
+    #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+    pub struct OutputContract {
+        #[serde(deserialize_with = "serde_helpers::deserialize_abi_string")]
+        pub abi: Vec<serde_json::Value>,
+        pub bin: String,
+        #[serde(rename = "bin-runtime")]
+        pub bin_runtime: String,
+    }
+
+    impl OutputContract {
+        fn keys() -> Vec<String> {
+            vec![
+                "abi".to_string(),
+                "bin".to_string(),
+                "bin-runtime".to_string(),
+            ]
+        }
+    }
+
+    impl TryFrom<OutputContract> for Contract {
+        type Error = SolcError;
+
+        fn try_from(output_contract: OutputContract) -> Result<Self, Self::Error> {
+            let contract = serde_json::json!({
+                "abi": output_contract.abi,
+                "evm": {
+                    "bytecode": {
+                        "object": output_contract.bin,
+                    },
+                    "deployedBytecode": {
+                        "object": output_contract.bin_runtime,
+                    },
+                },
+            });
+            let contract: Contract = serde_json::from_value(contract)?;
+            Ok(contract)
+        }
+    }
+
+    #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+    pub struct OutputJson {
+        pub contracts: HashMap<String, OutputContract>,
+    }
+
+    impl TryFrom<OutputJson> for CompilerOutput {
+        type Error = SolcError;
+
+        fn try_from(output_json: OutputJson) -> Result<Self, Self::Error> {
+            let contracts = output_json
+                .contracts
+                .into_iter()
+                .map(|(name, output)| output.try_into().map(|contract| (name, contract)))
+                .collect::<Result<HashMap<String, Contract>, _>>()?;
+            let contracts_raw = serde_json::to_value(&contracts)?;
+            let compiler_output = serde_json::json!(
+                {
+                    "contracts": {
+                        // TODO: give filename, if only 1 file was provided
+                        "": contracts_raw
+                    }
+                }
+            );
+            let compiler_output = serde_json::from_value(compiler_output)?;
+            Ok(compiler_output)
+        }
+    }
+}
+
+fn compiler_error(message: String) -> ethers_solc::artifacts::Error {
+    ethers_solc::artifacts::Error {
+        source_location: None,
+        secondary_source_locations: vec![],
+        r#type: "".to_string(),
+        component: "".to_string(),
+        severity: Severity::Error,
+        error_code: None,
+        message,
+        formatted_message: None,
+    }
+}
+
+pub fn compile_using_cli(
```

Comment:
Nit: personally, I believe it would be better to rename that function to `compile`, make `pub(crate) mod solc_cli` inside `mod.rs`, and use it as `solc_cli::compile` from the `compilers.rs`

Reply:
we have like 4 or 5 `compile` functions. don't you think it will confuse people?

Reply:
I actually believe that `compilers::compile` is excessive and сan just be directly inline inside `compilers::Compilers::compile`. So my idea was that after removal of `compilers::compile` we would have `solc_cli::compile` and `solc.compile` which does not seem actually confusing for me.